notebooks/rnn/lyft-rnn-lstm.py

```diff
@@ -381,76 +384,74 @@ def getTrainingSets(agents, limit):
 
         allTrainingSets.append(agentTrainingSets)
         pb.check(i)
-
+    pb.end()
+    
     print("len(allTrainingSets)", len(allTrainingSets))
     print("len(allTrainingSets[0])",len(allTrainingSets[0]), "\n")
 
     print("len(agentsOverLimit)",len(agentsOverLimit))
     print("len(agentsOverLimit[0]) - limit - 1",len(agentsOverLimit[0]) - limit - 1, "\n")
 
     print("totalNumberOfTrainingSets",totalNumberOfTrainingSets)
-    return allTrainingSets
+    return allTrainingSets, totalNumberOfTrainingSets
 
 
 # In[]:
 
 
-allTrainingSets = getTrainingSets(agentsOverLimit, limit)
+allTrainingSets, totalNumberOfTrainingSets = getTrainingSets(agentsOverLimit, limit)
 
 
 # In[]:
 
 
-def flattenTrainingSets(allTrainingSets):
-    allTrainingSetsFlattened_X = []
-    allTrainingSetsFlattened_Y = []
-    for allTrainingSet in allTrainingSets:
-        for trainingSet in allTrainingSet:
-            allTrainingSetsFlattened_X.append(trainingSet[0])
-            allTrainingSetsFlattened_Y.append(trainingSet[1])
-    print("len(allTrainingSetsFlattened_X)", len(allTrainingSetsFlattened_X))
-    return allTrainingSetsFlattened_X, allTrainingSetsFlattened_Y
+print(len(allTrainingSets))
+print(len(allTrainingSets[0]))
+print(len(allTrainingSets[0][0]))
+print(len(allTrainingSets[0][0][0]))
+print(len(allTrainingSets[0][0][0][0]))
 
 
 # In[]:
 
 
-allTrainingSetsFlattened_X, allTrainingSetsFlattened_Y = flattenTrainingSets(allTrainingSets)
+def flattenTrainingSets(allTrainingSets, totalNumberOfTrainingSets):
+    allTrainingSetsFlattened_X = np.empty((totalNumberOfTrainingSets, limit, len(allTrainingSets[0][0][0][0])))
+    allTrainingSetsFlattened_Y = np.empty((totalNumberOfTrainingSets, len(allTrainingSets[0][0][0][0])))
+    count = 0
+    for allTrainingSet in allTrainingSets:
+        for trainingSet in allTrainingSet:
+            allTrainingSetsFlattened_X[count] = np.array(trainingSet[0])
+            allTrainingSetsFlattened_Y[count] = trainingSet[1]
+            count = count + 1
+    print("len(allTrainingSetsFlattened_X)", len(allTrainingSetsFlattened_X))
+    return allTrainingSetsFlattened_X, allTrainingSetsFlattened_Y
 
 
 # In[]:
 
 
-def reshapeFlattenedTrainingSets(allTrainingSetsFlattened_X, allTrainingSetsFlattened_Y):
```

Comment:
Removed reshaping function since the previous step now outputs a numpy array so we don't have to reshape the list to a numpy array